corehq/motech/repeaters/models.py

```diff
@@ -975,6 +986,43 @@ def datasource_is_subscribed_to(domain, data_source_id):
             domain=domain, options={"data_source_id": data_source_id}
         ).exists()
 
+    # TODO: Test
+    def merge_records(self):
+        """
+        Merges updates to the same data source.
+
+        * Selects payloads for merging
+        * Registers a new repeat record for each group of payloads. The
+          new repeat record stores a list of payload IDs in its
+          ``payload_id`` field.
+        * Cancels the merged repeat records
+
+        ``Repeater.payload_doc()`` returns the merged payloads based on
+        the IDs in new repeat record's ``payload_id`` field.
+        """
+
+        # payload_id is a `models.CharField(max_length=255)` field. The
+        # payload_ids of (unmerged) repeat records are form and case
+        # IDs, which are 32 (`uuid4().hex`) or 36 (`str(uuid4())`)
+        # characters long. Assuming 36 + 1 character for a separator
+        # means we can get 6 payload IDs in a 255-character field.
+        num_records_to_merge = 6 * self.num_workers
```

Comment:
Maybe asking the obvious here, but this implies we will only be able to chunk up to 6 payloads, right?

(Thanks for the comment in the code, it's super helpful!)